### PR TITLE
[SPARK-30304][CORE]When the specified shufflemanager is incorrect, print the prompt.

### PR DESCRIPTION
**Pull request by sev7e0**
### What changes were proposed in this pull request?

During the instantiation of the specified `ShuffleManager`, if the configuration is wrong, whether the log can print some tips.

before:
```log
java.lang.ClassNotFoundException: hash
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at org.apache.spark.util.Utils$.classForName(Utils.scala:206)
	at org.apache.spark.SparkEnv$.instantiateClass$1(SparkEnv.scala:274)
	at org.apache.spark.SparkEnv$.create(SparkEnv.scala:338)
	at org.apache.spark.SparkEnv$.createDriverEnv(SparkEnv.scala:188)
	at org.apache.spark.SparkContext.createSparkEnv(SparkContext.scala:277)
	at org.apache.spark.SparkContext.<init>(SparkContext.scala:462)
	at org.apache.spark.SparkContext.<init>(SparkContext.scala:131)
	at org.apache.spark.SortShuffleSuite.$anonfun$new$1(SortShuffleSuite.scala:67)
	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
```

Now it's not just printing out stack exceptions, but also printing error prompt logs.

### Why are the changes needed?

It is convenient to quickly locate problems through logs.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Add test in `SortShuffleSuite`.

### CODE REVIEW

**Comment by maropu** [comment]
ok to test

File: core/src/main/scala/org/apache/spark/SparkEnv.scala [hunk 1]
```diff
@@ -334,7 +334,15 @@ object SparkEnv extends Logging {
...
     val shuffleMgrClass =
       shortShuffleMgrNames.getOrElse(shuffleMgrName.toLowerCase(Locale.ROOT), shuffleMgrName)
-    val shuffleManager = instantiateClass[ShuffleManager](shuffleMgrClass)
+    var shuffleManager: ShuffleManager = null
+    try {
+      shuffleManager = instantiateClass[ShuffleManager](shuffleMgrClass)
+    } catch {
+      case e: Throwable =>
+        logError(s"Init shuffle manager error. Maybe $shuffleMgrName " +
+          s"shuffle manager doesn't exist.")
```
Comment (maropu) [inline core/src/main/scala/org/apache/spark/SparkEnv.scala]: nit: drop `s` in the head.

Comment (sev7e0) [reply core/src/main/scala/org/apache/spark/SparkEnv.scala]: OK, it has been removed. Thank!

**Comment by maropu** [comment]
cc: @vanzin

**Comment by srowen** [comment]
Meh, maybe. This is already telling you the class you specified doesn't exist.

**Comment by sev7e0** [comment]
> Meh, maybe. This is already telling you the class you specified doesn't exist.

Well, as you said, it's just that there's no obvious mention in the log of where the error occurred.

**Pull request closed without merging**